js/packages/openinference-core/src/trace/trace_config/OpenInferenceSpan.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  Span,
+  SpanContext,
+  SpanStatus,
+  TimeInput,
+  Attributes,
+  Exception,
+  AttributeValue,
+  Link,
+} from "@opentelemetry/api";
+import { TraceConfig } from "./types";
+import { mask } from "./maskingRules";
+
+/**
+ * A wrapper around the OpenTelemetry {@link Span} interface that masks sensitive information based on the passed in {@link TraceConfig}.
+ */
+export class OpenInferenceSpan implements Span {
```

Comment:
I thought in python this is just called MaskedSpan? I think that might be simpler in name.

Also is there no base class to extend?

Reply:
It is, that name just felt very restrictive to me, but if we only need it to mask things i can change it.

There are several implementations of the Span interface, there isn't a base one to my knowledge. ONe of them might be a good candidate for extending. Not sure if we want to tie our selves to a specific implmentation or just compose the "MaskedSpan" with wahtever span teh tracer creates

Reply:
can take a look into this and tracer to see if there is a good implementation to extend